__tests__/unit/fact.test.js

```diff
@@ -0,0 +1,359 @@
+    ];
+    const sentence = getDefaultMsg(9999999999, "math", {});
+    expect(mathMsgs.includes(sentence)).toEqual(true);
+  });
+
+  test("default msg for type:'date' with no fact", function () {
+    const sentence = getDefaultMsg(1 / 1, "date", {});
+    expect(sentence).toBe(
+      "January 1st is the day that no newsworthy events happened."
+    );
+  });
+});
+
+describe("dataPairs", () => {
+  // dataPairs returns
+  // {"date": [Array], "math": [Array], "trivia": [Array], "year": [Array]}
+  // {'number': 1, string: '1'}
+
+  test("return object with keys 'date', 'math', 'trivia', 'year'", function () {
+    expect(Object.keys(dataPairs)).toEqual(["date", "year", "trivia", "math"]);
+  });
+
+  test("date contains data in correct format", function () {
+    let date = dataPairs["date"];
+    expect(date).toEqual(
+      expect.arrayContaining([
+        expect.objectContaining({ number: 1, string: "1" }),
+      ])
+    );
+  });
+
+  test("math contains data in correct format", function () {
+    let math = dataPairs["math"];
+    expect(math).toEqual(
+      expect.arrayContaining([
+        expect.objectContaining({ number: 1, string: "1" }),
+      ])
+    );
+  });
+
+  test("trivia contains data in correct format", function () {
+    let trivia = dataPairs["trivia"];
+    expect(trivia).toEqual(
+      expect.arrayContaining([
+        expect.objectContaining({ number: 1, string: "1" }),
+      ])
+    );
+  });
+
+  test("year contains data in correct format", function () {
+    let year = dataPairs["year"];
+    expect(year).toEqual(
+      expect.arrayContaining([
+        expect.objectContaining({ number: 1, string: "1" }),
+      ])
+    );
+  });
+});
+
+describe("filterObj()", () => {
+  test("return filtered object that has values for whitelisted keys", function () {
+    let filtered = filterObj(
+      {
+        hello: "world",
+        not: "this",
+      },
+      ["hello"]
+    );
+    expect(filtered).toEqual({ hello: "world" });
+  });
+
+  test("return empty object when whitelist is empty", function () {
+    let emptyWhitelist = filterObj(
+      {
+        hello: "world",
+        not: "this",
+      },
+      []
+    );
+    expect(emptyWhitelist).toEqual({});
+  });
+
+  test("return empty object when whitelist key is not in object", function () {
+    let notPresentKey = filterObj(
+      {
+        hello: "world",
+        not: "this",
+      },
+      ["whiskey"]
+    );
+    expect(notPresentKey).toEqual({});
+  });
+});
+
+describe("apiExtend()", () => {
+  test("filters object and extends with new data", function () {
+    let extend = apiExtend({ year: 2020, random: "lol" }, { newData: "wow" });
+    expect(extend).toEqual({ year: 2020, newData: "wow" });
+  });
+});
+
+describe("getFact()", () => {
+  test("return error object for invalid input type", function () {
+    let invalidType = getFact(1000, "sdasd", {});
```

Comment:
here too, would be nice if this call looked like:
```suggestion
    let invalidType = getFact({ number: 1000, type: "sdasd" });
```